src/main/scala/chiseltest/package.scala

```diff
@@ -205,7 +205,58 @@ package object chiseltest {
       case x => throw new LiteralTypeException(s"don't know how to peek $x")
     }
 
-    def peek(): T = peekWithStale(false)
+    protected def peekWithStaleBool(stale: Boolean): Boolean = x match {
+      case (x: Bool) =>
+        Context().backend.peekBits(x, stale) match {
+          case x: BigInt if x == 0 => false
+          case x: BigInt if x == 1 => true
+          case x => throw new LiteralTypeException(s"peeked Bool with value $x not 0 or 1")
+        }
+    }
+
+    protected def peekWithStaleBigInt(stale: Boolean): BigInt = x match {
+      case (x: UInt) => Context().backend.peekBits(x, stale)
+      case (x: SInt) => Context().backend.peekBits(x, stale)
+    }
+
+    protected def peekWithStaleDec(stale: Boolean): BigDecimal = x match {
+      case (x: FixedPoint) => {
+        val multiplier = BigDecimal(2).pow(x.binaryPoint.get)
+        BigDecimal(Context().backend.peekBits(x, stale)) / multiplier
+      }
+    }
+
+    protected def peekWithStaleVecInt(stale: Boolean): Seq[BigInt] = x match {
+      case (x: Vec[UInt]) => {
+        val elementValueFns = x.getElements.map { case elt: Data =>
+          elt.peekWithStaleBigInt(stale)
+        }
+        Seq(elementValueFns: _*)
+      }
+      case (x: Vec[SInt]) => {
+        val elementValueFns = x.getElements.map { case elt: Data =>
+          elt.peekWithStaleBigInt(stale)
+        }
+        Seq(elementValueFns: _*)
+      }
+    }
+
+    protected def peekWithStaleVecBool(stale: Boolean): Seq[Boolean] = x match {
+      case (x: Vec[Bool]) => {
+        val elementValueFns = x.getElements.map { case elt: Data =>
+          elt.peekWithStaleBool(stale)
+        }
+        Seq(elementValueFns: _*)
+      }
+    }
+
+    def peek():           T = peekWithStale(false)
+    def peekBoolean():    Boolean = peekWithStaleBool(false)
+    def peekBigInt():     BigInt = peekWithStaleBigInt(false)
+    def peekInteger():    Integer = peekWithStaleBigInt(false).toInt
```

Comment:
```suggestion
    def peekInt():    Int = peekWithStaleBigInt(false).toInt
```
I would stick with `Int` which is the Scala type, not `Integer` which is actually a Java type.

That being said, I agree with Kevin that I don't see value in `Int` over `BigInt`. In my experience, `Int` just leads to grief when accidentally overflowing so it might be best to just stick with BigInt. The implicit conversion from Int to BigInt makes comparisons and stuff work very intuitively.

Reply:
Awesome, so can I keep the `peekInt` function but returning BigInt?
Also go back to `Bool` instead of `Boolean` for shortness?

Reply:
Better now:

```scala
    def peek():        T = peekWithStale(false)
    def peekBool():    Boolean = peekWithStaleBool(false)
    def peekInt():     BigInt = peekWithStaleBigInt(false)
    def peekDec():     BigDecimal = peekWithStaleDec(false)
    def peekVecInt():  Seq[BigInt] = peekWithStaleVecInt(false)
    def peekVecBool(): Seq[Boolean] = peekWithStaleVecBool(false)
```

Reply:
OK, my mistake on ```Integer```. And agree that ```BigInt``` is enough. But then I would still call it ```peekBigInt``. And also ```peekBoolean``` as ```Bool``` is a Chisel type. Furthermore, what does ```peekVec``` return? I guess not a ```Vec```., as that is a Chisel type. Maybe we do not really need the peeking of a Vec, as one can index in the tester and then just use the ```peekBigInt``` on that index. I do not see a big value on getting a ```Seq``` back.

Reply:
News about this? Should I keep then as they are now?